---
title: "AllrRNAfigures"
author: "Saaz Sakrikar, Amy Schmid"
date: "9/14/2021"
output:
  html_document: default
---

Load required  packages

```{r packages}
library(forcats)
library(ggplot2)
library(readxl)
library(reshape2)
library(gridExtra)
```


Load data for each figure

```{r load_data}

#data for figure 1 - no of publications for "archaea" vs "rna-seq" vs "archaea rna-seq"
Archaea_RNA_pubs=read_excel("TableS2_Archaea_RNApubs-AKS.xlsx")

#data for figure 2 - old ribozero on hbt sal and hfx vol
Old_rzero=read_excel("TableS1_rRNAresults-AKS-2022-02-24.xlsx",sheet = "Fig2_rzero")

#data for fig 3 - alternative  methods  in hbt sal
Alt_hbt=read_excel("TableS1_rRNAresults-AKS-2022-02-24.xlsx",sheet="Fig3_Hbtsal")

#data for fig 4 - effect of RNAse digestion on removal efficiency
Rnase_time=read_excel("TableS1_rRNAresults-AKS-2022-02-24.xlsx",sheet="Fig4_time")

#data for fig 5 - comparing custom probe methods
probemethods=read_excel("TableS1_rRNAresults-AKS-2022-02-24.xlsx",sheet="Fig5_probemethods")

#data for fig 6 - sitools panarchaea kit across species
panarchaea_data=read_excel("TableS1_rRNAresults-AKS-2022-02-24.xlsx",sheet="Fig6_pan")

#data for fig 7A - effect of rRNA removal on reads mapping to genes
#Alignedreads=read_excel("TableS1_rRNAresults-AKS.xlsx",sheet="Fig7A_alignedreads")

#data for fig 7B - effect of rRNA removal on low-count genes
reads_vs_rrna=read_excel("TableS1_rRNAresults-AKS-2022-02-24.xlsx",sheet="Fig7B_rRNAeffect")

#data for fig 8a - correlations between hbt sal avg reads in different methods/batches
Hbt_correlations=read_excel("TableS5_correlations.xlsx",sheet="Hbt_avg")

#data  for fig 8b - correlations between hfx vol avg reads from different methods/batches
Hvo_correlations=read_excel("TableS5_correlations.xlsx",sheet="Hvo_avg")

#data for fig s1a - identity of custom probe sequences to Hbt rRNA sequences
probespecificity=read_excel("TableS3_probespecificity-AKS-2022-02-25.xlsx",sheet="Fig4-probespecificity")

#data for fig s1b - 16S vs 23S removal by custom probes in Hbt sal
Hbtprobeselectivity=read_excel("TableS3_probespecificity-AKS-2022-02-25.xlsx",sheet="Fig4-NEBcustomselectivity")

#data for fig s2 - intergenic reads comparison
Panarch_gDNA=read_excel("TableS4_gDNA.xlsx",sheet="FigS2_Speciescomp")


```

Figure showing slowdown in archaeal RNA-Seq research after withdrawal of Ribozero kit
```{r publications}
#Fig 1 - processing the data (melting) and then plotting
Archaea_RNA_pubs=Archaea_RNA_pubs[,-1]
Archaea_RNA_pubs_melted=melt(Archaea_RNA_pubs,id.vars="Year")
pdf("../Figures/Fig1-pubsovertime.pdf",height=4,width=8)
ggplot(Archaea_RNA_pubs_melted,aes(x=Year,y=value,color=variable)) +
  geom_line(size=1) +
  geom_point() + 
  theme_bw() +
  scale_y_log10() + 
  ylab("Log # publications") + 
  xlab ("Year") + 
  geom_vline(xintercept = 2018, linetype = "dotted") +
  labs (linetype = "Publication topic")
dev.off()

```


All figures showing  rRNA % left after each removal method  is used.
```{r rRNApics}
#Fig 2
#pdf("../Figures/Fig2-OldRZ.pdf",height=4,width=8)
ggplot(Old_rzero,aes(x=fct_inorder(Species),y=RRNA,color=Species,fill=Species)) +
  geom_dotplot(binaxis="y",stackdir="center") + 
  stat_summary(fun=median,geom="crossbar",size=0.4, width = 0.4) +
  scale_fill_manual(values=c("coral","steelblue")) +
  scale_color_manual(values=c("coral","steelblue")) +
  theme_bw() +
  ylim(c(0,100)) +
  ylab ("% rRNA remaining") + 
  xlab ("Species")
  #ggtitle("Fig 2 - rRNA removal using discontinued ribozero kit")
#dev.off()

```

```{r}
#Fig 3
#pdf("../Figures/Fig3-Hbtremoval.pdf",height=4,width=8)
ggplot(Alt_hbt,aes(x=fct_inorder(Method),y=RRNA,fill=Method,color=Method)) +
  geom_dotplot(binaxis="y",stackdir="center") +
  stat_summary(fun=median,geom="crossbar",size=0.4, width = 0.5) +
  scale_fill_manual(values=c("coral2","coral1","coral4","coral3")) +
  scale_color_manual(values=c("coral2","coral1","coral4","coral3")) +
  theme_bw() +
  ylim(c(0,100)) +
  ylab ("% rRNA remaining") + 
  xlab ("Method")
  #ggtitle("Fig 3 - alternative methods tested on Hbt salinarum")
#dev.off()
```

```{r}
#Fig 4 

Rnase_time_Hbt=Rnase_time[Rnase_time$Species=="Halobacterium salinarum",]

#Fig 4A
#pdf("../Figures/Fig4A-HbtRNAsetime.pdf",height=4,width=8)
pa <- ggplot(Rnase_time_Hbt,aes(x=fct_inorder(Methodbatch),y=RRNA,fill=Method,color=Method)) +
  geom_dotplot(binaxis="y",stackdir="center") +
  stat_summary(fun=median,geom="crossbar",size=0.4, width = 0.3) +
  scale_fill_manual(values=c("coral1","coral4")) +
  scale_color_manual(values=c("coral1","coral4")) +
  theme_bw() +
  ylim(c(0,100)) +
  xlab ("Digestion time (min)") + 
  ylab ("% rRNA remaining")
  #ggtitle("Fig 4A - effect of RNase digestion on rRNA removal in Hbt sal")
#dev.off()

#Fig 4B
Probes_melted=melt(probespecificity,id.vars="Similarity_range")
#pdf("Probespecificity.pdf",height=4,width=8)
pb <- ggplot(Probes_melted,aes(x=fct_inorder(Similarity_range),y=value,color=variable)) +
  scale_color_manual(values=c("black","grey")) +
  theme_bw() +
  geom_point(size=2) +
  ylim(c(0,50)) +
  xlab ("Similarity range") +
  ylab ("% Probes")
  
#dev.off()

#Fig 4C
Hbtprobespecificity_melted=melt(Hbtprobeselectivity,id.vars="Method")
#pdf("Proberesults.pdf",height=4,width=8)
pc <- ggplot(Hbtprobespecificity_melted,aes(x=fct_inorder(Method),y=value,fill=Method,color=Method)) +
  geom_dotplot(dotsize=0.9,binaxis="y",stackdir="center") +
  stat_summary(fun=median,geom="crossbar",size=0.4, width = 0.5) +
  facet_wrap(~variable) +
  scale_fill_manual(values=c("coral3","coral2","coral4")) +
  scale_color_manual(values=c("coral3","coral2","coral4")) +
  theme_bw() +
  geom_point() + 
  ylab ("% rRNA remaining") + 
  xlab ("Method")
#dev.off()

#export graphs as multpanel figure
# pdf("../Figures/fig4-probespecificty.pdf", width = 5)
# grid.arrange(pa, pb, pc, ncol = 1)
# dev.off()

```

```{r}
#Fig 5 - siTOOLS and NEB equally effective in removal of rRNA from HVO with HVO specific probes.

#pdf("Probevspecies.pdf",height=4,width=8)
#ggplot(probemethods,aes(x=fct_inorder(Methodbatch),y=RRNA,fill=Method,color=Method))+geom_dotplot(binaxis="y",stackdir="center")+stat_summary(fun=median,geom="crossbar",size=0.4)+scale_fill_manual(values=c("lightcoral","steelblue","orchid"))+scale_color_manual(values=c("lightcoral","steelblue","orchid"))+theme_bw()+ylim(c(0,100))+ggtitle("Fig 5 - methods using custom-designed probes targeted to Hfx vol")
#dev.off()

Hvoprobemethods=probemethods[probemethods$Species=="HVO",]

#pdf("../Figures/Fig5-HVO-NEB-rP.pdf",height=4,width=8)
ggplot(Hvoprobemethods,aes(x=fct_inorder(Method),y=RRNA,fill=Method,color=Method)) +
  geom_dotplot(binaxis="y",stackdir="center") +
  stat_summary(fun=median,geom="crossbar",size=0.4, width = 0.4) +
  scale_fill_manual(values=c("dark blue","steelblue")) +
  scale_color_manual(values=c("dark blue","steelblue")) +
  theme_bw() +
  ylim(c(0,0.02)) +
  xlab ("Method") +
  ylab ("% rRNA remaining")
 # ggtitle("Fig 5 - methods using custom-designed probes - inset of Hfx vol results")
#dev.off()
```

```{r}
#Fig 6 - Panarchaea results including inset

#pdf("Panarchaea.pdf",height=4,width=8)
#ggplot(panarchaea_data,aes(x=fct_inorder(Species),y=RRNA,fill=Species,color=Species)) +
 # geom_dotplot(binaxis="y",stackdir="center") +
#  stat_summary(fun=median,geom="crossbar",size=0.4) +
 # scale_fill_manual(values=c("lightcoral","black","orchid","steelblue","forestgreen")) +
  #scale_color_manual(values=c("lightcoral","black","orchid","steelblue","forestgreen")) +
  #theme_bw() + 
  #ylim(c(0,100)) +
  #ggtitle("Fig 6 - Panarchaea tested across halophiles")
#dev.off()

#pdf("../Figures/Fig6-Panarchaea.pdf",height=4,width=8)
ggplot(panarchaea_data,aes(x=fct_inorder(Species),y=RRNA,fill=Species,color=Species)) +
  geom_dotplot(binaxis="y",stackdir="center") +
  stat_summary(fun=median,geom="crossbar",size=0.4, width = 0.4) +
  scale_fill_manual(values=c("coral","grey50","orchid","steelblue")) +
  scale_color_manual(values=c("coral","grey50","orchid","steelblue")) +
  theme_bw() +
  ylim(c(0,8)) +
  xlab ("Species") +
  ylab ("% rRNA remaining")
  #ggtitle("Fig 6 - Panarchaea tested across halophiles-inset")
#dev.off()

```




Correlations between methods

```{r correlations,warning=FALSE}
#Fig 7a
#pdf("../Figures/Hbtcorrelations.pdf",height=4,width=8)
hbt.cors<-pairs(Hbt_correlations[,2:6],log="xy",cex=0.25)
#dev.off()

#Fig 7b
#pdf("../Figures/HVOcorrelations.pdf",height=4,width=8)
hvo.cors<-pairs(Hvo_correlations[,2:5],log="xy",cex=0.25)
#dev.off()




```

Data on reads aligning to genes depending on rRNA removal.
```{r Totalreads}

#Fig 8
#pdf("../Figures/Fig8-lowcountgenes.pdf",height=4,width=5)
ggplot(reads_vs_rrna) +
  geom_point(size=2,shape = 1, aes(x=rRNA,y=Zerocountgenes,color=Method)) +
  theme_bw() +
  geom_point(size=2,shape=19,aes(x=rRNA,y=Underfivecountgenes,color=Method)) +
  ylab("Number of lowly expressed genes") +
  xlab("% rRNA remaining") +
  scale_color_manual(values=c("grey60","grey40","grey80","black")) #+
  #ggtitle("Fig 7B - Comparing genes at 0 and <5 counts from different Hbt sal removal methods")
#dev.off()
```
